### PR TITLE
Added auth complete flow

### PR DESCRIPTION
**Pull request by kapeiharshad**
1)Added auth api for users and admin.
2)Added login api with jwt verification.
3) Added change password api.
4) Added forget password api which will send email for verification.
5) Added its swagger and test cases.
6) Updated configuration of test and util file.

test
auth_swagger

### CODE REVIEW

File: models/userToken.model.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      type: String,
+      required: true,
+    },
+    expiryTime: {
+      type: Date,
+      required: true,
+    },
+    type: {
+      type: String,
+      enum: ['accessToken', 'refershToken', 'randToken'],
```
Comment (kapeiharshad) [inline models/userToken.model.js]: please update refreshToken name properly.

File: helpers/jwt.helper.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    secret = process.env.JWT_TOKEN_SECRET ? process.env.JWT_TOKEN_SECRET : '';
+    if (!expiry || !secret) {
+      return false;
+    }
+    return jwt.sign(payload, secret, { expiresIn: expiry });
+  }
+
+  static async getRequestToken(req) {
+    let token = null;
+    if (req.headers.authorization) {
```
Comment (pratikgupta9) [inline helpers/jwt.helper.js]: iski zarurat lagg rahi hai yaha (req.headers && req.headers.authorization)

File: services/auth.service.js [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+          },
+          { upsert: true },
+        );
+        if (
+          saveUserToken &&
+          (saveUserToken.modifiedCount || saveUserToken.upsertedCount)
+        ) {
+          return {
+            success: true,
+            token: token,
```
Comment (pratikgupta9) [inline services/auth.service.js:50]: statusCode=200 missing

File: services/auth.service.js [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+      }
+      return {
+        success: false,
+        statusCode: 401,
+        msg: 'Email or password is incorrect!',
+      };
+    } catch (error) {
+      logger.error('From login api error', { errorMsg: error });
+      return {
+        success: false,
```
Comment (pratikgupta9) [inline services/auth.service.js:64]: statusCode ki zaraurat hai yaha then add karna?

File: services/users.service.js [hunk 5]
```diff
@@ -153,7 +154,7 @@ class UserService {
       } else {
         return {
           success: false,
-          statusCode: 404,
+          statusCode: 400,
           msg: 'User not founded.',
```
Comment (pratikgupta9) [inline services/users.service.js]: founded ko found kar na isme

File: services/users.service.js [hunk 6]
```diff
@@ -182,7 +183,7 @@ class UserService {
       } else {
         return {
           success: false,
-          statusCode: 404,
+          statusCode: 400,
           msg: 'User not founded or deleted unsuccessfully.',
```
Comment (pratikgupta9) [inline services/users.service.js]: founded ko found karna bas

File: services/auth.service.js [hunk 7]
```diff
@@ -0,0 +1,284 @@
...
+          success: false,
+          statusCode: 400,
+          msg: 'Must contains token on request body or header',
+        };
+      }
+      await UserToken.deleteMany({ token: headerToken });
+      if (body.hasOwnProperty('token')) {
+        await UserToken.deleteMany({ token: body.token });
+      }
+      return {
```
Comment (pratikgupta9) [inline services/auth.service.js:276]: statusCode missing in this function

File: services/auth.service.js [hunk 8]
```diff
@@ -0,0 +1,284 @@
...
+            ),
+            locals: {
+              emailBody,
+              subject: emailTemplate.subject,
+            },
+          };
+          new Promise(async () => {
+            await Mailer.sendEmail(mailData);
+          });
+          return {
```
Comment (pratikgupta9) [inline services/auth.service.js:169]: statusCode add karna

File: services/auth.service.js [hunk 9]
```diff
@@ -0,0 +1,284 @@
...
+      } else {
+        return {
+          success: false,
+          statusCode: 400,
+          msg: 'Reset password unsuccessfully',
+        };
+      }
+    } catch (error) {
+      logger.error('From resetPassword api error', { errorMsg: error });
+      return {
```
Comment (pratikgupta9) [inline services/auth.service.js:255]: statusCode add karna

File: services/auth.service.js [hunk 10]
```diff
@@ -0,0 +1,284 @@
...
+        const updatePassword = await User.updateOne(
+          { _id: mongoose.Types.ObjectId(userToken.userId) },
+          { password: hash },
+        );
+        if (updatePassword && updatePassword.modifiedCount) {
+          new Promise(async () => {
+            await UserToken.deleteOne({ type: 'randToken', token: body.token });
+          });
+          return {
+            success: true,
```
Comment (pratikgupta9) [inline services/auth.service.js:240]: statusCode add karna

File: services/auth.service.js [hunk 11]
```diff
@@ -0,0 +1,284 @@
...
+          statusCode: 400,
+          msg: 'Must contains token on request body or header',
+        };
+      }
+      await UserToken.deleteMany({ token: headerToken });
+      if (body.hasOwnProperty('token')) {
+        await UserToken.deleteMany({ token: body.token });
+      }
+      return {
+        success: true,
```
Comment (pratikgupta9) [inline services/auth.service.js:277]: statusCode missing

File: services/auth.service.js [hunk 12]
```diff
@@ -0,0 +1,284 @@
...
+        msg: 'Logout successfully',
+      };
+    } catch (error) {
+      logger.error('From logout api error', { errorMsg: error });
+      return {
```
Comment (pratikgupta9) [inline services/auth.service.js:282]: statusCode missing

File: services/auth.service.js [hunk 13]
```diff
@@ -0,0 +1,284 @@
...
+    } catch (error) {
+      logger.error('From forgotPassword api error', { errorMsg: error });
+      return {
+        success: false,
+        msg: error,
+      };
+    }
+  }
+
+  static async resetPasswordVerify({ body }) {
```
Comment (pratikgupta9) [inline services/auth.service.js:194]: statusCode missing in this function

**Review by pratikgupta9 — CHANGES_REQUESTED** [review]

File: services/auth.service.js (hunk 3 above)
Comment (kapeiharshad) [reply services/auth.service.js:50]: by default sucess has 200 response

File: models/userToken.model.js (hunk 1 above)
Comment (kapeiharshad) [reply models/userToken.model.js]: done

File: helpers/jwt.helper.js (hunk 2 above)
Comment (kapeiharshad) [reply helpers/jwt.helper.js]: updated done.

File: services/auth.service.js (hunk 4 above)
Comment (kapeiharshad) [reply services/auth.service.js:64]: done

File: services/users.service.js (hunk 5 above)
Comment (kapeiharshad) [reply services/users.service.js]: done

File: services/users.service.js (hunk 6 above)
Comment (kapeiharshad) [reply services/users.service.js]: done

File: services/auth.service.js (hunk 9 above)
Comment (kapeiharshad) [reply services/auth.service.js:255]: done

File: services/auth.service.js (hunk 10 above)
Comment (kapeiharshad) [reply services/auth.service.js:240]: no need to add

File: services/auth.service.js (hunk 11 above)
Comment (kapeiharshad) [reply services/auth.service.js:277]: no need to add

File: services/auth.service.js (hunk 12 above)
Comment (kapeiharshad) [reply services/auth.service.js:282]: done

File: services/auth.service.js (hunk 13 above)
Comment (kapeiharshad) [reply services/auth.service.js:194]: no need to add

**Pull request merged**